src/core_crypto/matrix.rs

```diff
@@ -43,6 +42,16 @@ where
         self.as_mut()[row_idx].as_mut()[column_idx] = val;
     }
 
+    /*  Ideally, you want the return type to be MatrixMut, However this is not doable because of object safety
+    However, when R is a vector or align vector, the result will always be a matrixmut because we explicitly implemented MatrixMut
+    for them.
+
+    Note the postion idx is included in the second half
+    */
+    fn split(&mut self, idx: usize) -> (&mut [<Self as Matrix>::R], &mut [<Self as Matrix>::R]) {
```

Comment:
A small thing.

Can we change this to `split_at_row`? Much clearer.

Reply:
yes you are right